js/compiler/compiler.go

```diff
@@ -204,3 +227,40 @@ func (b *babel) Transform(logger logrus.FieldLogger, src, filename string) (stri
 	}
 	return code, &srcMap, err
 }
+
+// Pool is a pool of compilers so it can be used easier in parallel tests as they have their own babel.
+type Pool struct {
```

Comment:
Hmm if we're not using this internally in k6, and we probably won't ever need it, why don't we implement it in `js/compiler/compiler_test.go`? TC39 tests (and `js/http` tests in the future) will still be able to import it and use it, but it won't "pollute" k6 release code until we actually need it.

Reply:
It is literally 20 lines of code that doesn't add any dependency and having `js/http` depend on `tc39` seems ... like not so great idea ... 

But if you both want it so much I guess I am gonna move it :shrug:

Reply:
> having `js/http` depend on `tc39` seems ... like not so great idea ...

You didn't read my suggestion correctly - I don't want to move it in `tc39`, I want it to stay in the exact same package it currently is, just in a `_test.go` file. This way, other test packages will still be able to import it without any changes, but it won't be part of the normal k6 release code. And it's not that it will increase the binary size, I just don't want to have something in the codebase that is meant just for tests, that gets confusing very quickly, especially for newcomers.

Reply:
> I want it to stay in the exact same package it currently is, just in a _test.go file. 

and 

> This way, other test packages will still be able to import it without any changes,

are contradicting AFAIK - you can *not* import things from `_test.go` files of one package in another `_test.go` files as ... the original package is still not compiled with the tests ;) . see https://stackoverflow.com/questions/31794141/can-i-create-shared-test-utilities

Reply:
so even if I move it to tc39 ... it will still need to be in a *non* _test.go file which IMO will be even more confusing.

Also to reiterate the *whole* codebase around caching babel in this packages is about tests specifically as it makes no sense in the case of a running k6 where there will be only 1 compiler and compiling and transpiling source codes one after the other

Reply:
Ah, yeah... :face_with_head_bandage: I thought this wasn't a problem because I remember that the libraries were in `_test.go` file in the cloud integration tests, and while that's true, we don't have multiple modules there :man_facepalming: :sweat_smile:

Reply:
Fair enough, objections withdrawn :sweat_smile: